src/withLocation.test.js

```diff
@@ -0,0 +1,59 @@
+import '../test/jest/__mock__';
+import { MemoryRouter } from 'react-router';
+import { cleanup, fireEvent, render, screen } from '@testing-library/react';
+import withLocation from './withLocation';
+
+const mockGoto = jest.fn();
+const mockUpdateLocation = jest.fn();
+const mockGetParams = jest.fn();
+const mockGetSearchParams = jest.fn();
+const defaultProps = {
+  location: {
+    search: '?filters=test1&query=test2&sort=test3&qindex=test',
+    pathname: '/testPathName',
+  },
+  history: {
+    push: jest.fn()
+  },
+  updateLocation: mockUpdateLocation,
+  goTo: mockGoto,
+  getParams: mockGetParams,
+  getSearchParams: mockGetSearchParams
```

Comment:
These mocks are only used here so there's no reason to define them separately/globally. Just assign them the way you do `defaultProps.history.push`.

Reply:
changes are done as you suggested.